ni-oe-init-build-env

```diff
@@ -21,10 +21,22 @@ BITBAKEDIR=${BITBAKEDIR:-${SCRIPT_ROOT}/sources/bitbake}
 BUILDDIR=${BUILDDIR:-${SCRIPT_ROOT}/build}
 
 BB_ENV_EXTRAWHITE="${BB_ENV_EXTRAWHITE} \
+	BB_NUMBER_THREADS \
 	ENABLE_BUILD_TAG_PUSH \
 	GIT_REPODIR \
 "
 
+# Users can assert the maximum number of concurrent bitbake threads via the
+# BB_NUMBER_THREADS variable, or let this script calculate a reasonable value
+# based on the number of cores.
+if [ -z "$BB_NUMBER_THREADS" ]; then
+	echo "INFO: BB_NUMBER_THREADS not set. Calculating..."
+	# num_cores * 2 is the recommended value from the yocto manual
+	BB_NUMBER_THREADS=$(($(nproc) * 2))
```

Comment:
If `nproc` is not found, wouldn't it be better to print, and continue with OE's default value?

Reply:
I really can't imagine there will be a scenario where a user *intentionally* doesn't have nproc in the PATH. I'd expect that the user would rather that the script fail and indicate that something is wrong. Can you think of a reasonable host setup that doesn't have nproc?

Reply:
`nproc` is part of GNU coreutils, which means you'd probably have to be on a distro with a busybox-based userland to not have it, or on a non-Linux system (macOS does not have it, I dunno about the BSDs), and I think if you're in that situation you're in a place that upstream OE doesn't support given the [build host requirements](https://www.yoctoproject.org/docs/latest/mega-manual/mega-manual.html#brief-compatible-distro).

Reply:
> Can you think of a reasonable host setup that doesn't have nproc?

No idea :). I'm just saying the error handling can be better. Instead of a nproc not found error, maybe just tell the user to set the BB_NUMBER_THREADS var. But maybe it's all a NOOP given Brandon's statement above.